v22.2/backup-validation.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Backup Validation
+summary: Ensure that your backups are restorable with backup validation.
+toc: true
+docs_area: manage
+---
+
+{% include_cached new-in.html version="v22.2" %} CockroachDB provides backup validation tools to ensure that backups you have in storage are restorable. You can use the following option patterns with [`SHOW BACKUP`](show-backup.html) and [`RESTORE`](restore.html) to validate a [cluster](backup.html#backup-a-cluster), [database](backup.html#backup-a-database), or [table](backup.html#backup-a-table-or-view) backup. The three options result in increasing levels of backup validation:
+
+1. `SHOW BACKUP ... WITH check_files`: Check that all files belonging to a backup are in the expected location in storage. See [Validate backup files](#validate-backup-files) for an example.
+1. `RESTORE ... WITH schema_only`: Restore a backup to verify that it is valid without restoring any user table data. See [Validate the restore of a backup](#validate-the-restore-of-a-backup) for an example.
+1. `RESTORE ... WITH schema_only, verify_backup_table_data`: Run a `schema_only` restore **and** have the restore read all user data from external storage, verify checksums, and discard the data before writing it to disk. To use `verify_backup_table_data`, you must include `schema_only` in the statement. See [Validate the restore of backup table data](#validate-the-restore-of-backup-table-data) for an example.
+
+The options that give the most validation coverage will increase the runtime of the check. That is, `verify_backup_table_data` will take a longer time to validate a backup compared to `check_files` or `schema_only` alone. Despite that, each of these validation options provide a quicker way to validate a backup over running a "regular" restore.
```

Comment:
Could we add here that restore with schema_only will provide the most bang for their buck ?

Reply:
I'd be careful about any "bang for buck" arguments. restore with schema_only doesn't necessarily check all of the user's backup files. If the user wants to run restore with_schema_only, they should also run show backup with check_files.  Kathryn wrote this below.

Reply:
What do you both think about around here (before the examples) adding a note box that calls out that object locking can protect the validity of backups? Feels at least semi-related as a "prevention" step?

Reply:
I'll leave this for now. We can revisit in the future if it comes up.